### ISSUE TITLE
bstables: Bulk Edit in Items.

### ISSUE THREAD

**Issue by odiea** [jekkos/opensourcepos#432]
With bootstrapui have you noticed that if no items are selected for bulk edit  and you select the button the warning message appears then the Bulk Edit form shows anyway.

**Comment by jekkos** [comment]
I'll revirew that part as I'm changing the code that handles this part anyway

**Comment by jekkos** [comment]
See also #293

**Comment by jekkos** [comment]
Bulk edit button is disabled now if no items are selected. This change is in bootstrap-tables branch. Can someone verify/close?

**Comment by odiea** [comment]
the table shows just fine and all 3 buttons are blocked. When you place a mark in the top check box to select all the rows, all 3 are still blocked. If you select just an item and then the top check box it appears to work but the edit form does not show.

**Comment by jekkos** [comment]
Issue with (un)select all should be resolved now. Buttons should be enabled/disabled properly. Also found the edit form to behave ok now.

**Comment by daN4cat** [comment]
With bstable I noticed that if I select 50 or 100 rows and do a bulk edit of let say the supplier of all the rows the web browser gets stuck. With 50 rows after a while the browser comes back to function properly.
All works fine with 20 rows for instance.

**Comment by daN4cat** [comment]
There is this issue that many row refreshes are pending and if you updated 100 rows the browser page gets stuck and from the inspector you can see that there are a number of pending GET requests.

Somehow there should be a general upgrade and one refresh for the batch at the end. The way is done now is too slow.

**Comment by daN4cat** [comment]
@jekkos this is the issue about bulk edit. It used to be another issue but since we changed to bstables it changed and the last problem is the upgrade of 100 of rows that blocks the web browser.

**Comment by jekkos** [comment]
Ok I reworked the refresh routine a bit. Did some testing on most of the functionality that should be tested. Row highlighting works now for all rows after bulk update is done. Also update in the dom is done using one request afterwards now (instead of one request per row). I haven't tested performance with a lot of rows but I guess it can't be worse.

**Comment by daN4cat** [comment]
ok, when available I'll test it and let you know.

**Comment by daN4cat** [comment]
@jekkos something is broken with the last change, when trying to do a bulk edit in Items I keep getting:

![image](https://cloud.githubusercontent.com/assets/636048/15582614/ca47fbd2-2369-11e6-9efa-14535456cbc9.png)

**Comment by daN4cat** [comment]
That's just with 20 rows

**Comment by daN4cat** [comment]
Bulk Edit of 100 rows seems to never comes back, the web browser gets stuck.
You should try that test scenario.

**Comment by jekkos** [comment]
OK think we saw this issue before but wasnt able to fix it. Might better to disable buttons while loading

**Comment by jekkos** [comment]
I think I found why this modal isn't loading properly Don't try to click the bulk button on it's icon, I think it breaks the dialog (will submit a fix for that)

**Comment by jekkos** [comment]
I have set up a db with more items now will do some more testing tormorrow. At first sight it's not the animation that's causing the lag (I disabled them and still same effect).. I'll try to use the profiler to see what's going on.

Or we might just consider to refresh the whole table after an upper update limit is exceeded.

**Comment by jekkos** [comment]
![screenshot from 2016-05-26 22 57 36](https://cloud.githubusercontent.com/assets/576234/15589926/486dd808-2395-11e6-9584-16a752b696a6.png)

**Comment by daN4cat** [comment]
Not sure I can read the picture properly. What's that saying?

**Comment by daN4cat** [comment]
BTW, I could see the issue when clicking the button icon. That's weird....

**Comment by jekkos** [comment]
After playing with the profiler conclusion is that bstables  is rebuilding the whole table body after every call to updateRowByUniqueId. At first sight there does not seem to be a bulk update functionality present. I guess this direct DOM update is only meant to be used to update one or several rows. 

Anyway I have just reverted to doing a full table refresh if the update row count is more than 10. Afterwards there's the highlight animation which is same when adding a new row.

The glyph error was due to a wrong selector in the parents() of that node, it should have searched for links and buttons instead of just l inks.
